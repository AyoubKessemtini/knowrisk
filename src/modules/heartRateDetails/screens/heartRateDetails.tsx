import React, { useEffect, useState } from 'react';
import { ActivityIndicator, StyleSheet, View } from 'react-native';
import { Screen } from '@components/Screen';
import { Header } from '@components/Headers/Header.tsx';
import { CText } from '@components/CText.tsx';
import { HeartRateAVGCard } from '@modules/heartRateDetails/components/heartRAteAVGCard.tsx';
import { core } from '@config/Configuration.ts';
import { RootState, useAppSelector } from '@store/index.ts';

interface HeartRateDetails {
  date: string;
  heartRate: number;
  heartRateVariability: number;
  day: string;
}

export const HeartRateDetailsScreen: React.FC = () => {
  const [heartRate, setHeartRate] = useState<HeartRateDetails[]>([]);
  const [todayData, setTodayData] = useState<HeartRateDetails | null>(null);
  const today: string = new Date().toISOString().split('T')[0];
  const { hr } = useAppSelector((state: RootState) => state.bleData);
  const [loading, setLoading] = useState<boolean>(true);

  useEffect(() => {
    const fetchHeartRateData = async () => {
      try {
        setLoading(true);
        const res = await core.getHeartRateWeeklyData.execute();
        const heartRateData: HeartRateDetails[] = res['value'];

        // Find today's data
        const todayData =
          heartRateData.find((data) => data.date === today) || null;

        setHeartRate(heartRateData);
        setTodayData(todayData);
        setLoading(false);
      } catch (error) {
        setLoading(false);
        console.error('Error fetching heart rate data:', error);
      }
    };

    fetchHeartRateData(); // Call the function
  }, [today]);

  return (
    <Screen
      fullscreen
      withoutTopEdge
      noHorizontalPadding
      containerStyles={styles.container}
    >
      <Header
        hasBackButton
        useCustomBackButton
        text="common.heartRate"
        textColor="black"
      />
      {loading && (
        <ActivityIndicator
          style={{ marginTop: 200 }}
          size="large"
          color="#0000ff"
        />
      )}
      {!loading && (
        <View style={styles.wrapper}>
          <CText size={'lg_light'}>Today</CText>
          <HeartRateAVGCard
            title={'Realtime Heart rate'}
            dateTime={'Now'}
            value={hr}
          />
          {todayData && (
            <>
              {todayData.heartRate && (
                <HeartRateAVGCard
                  title={'Heart rate average'}
                  dateTime={'Today'}
                  value={String(todayData.heartRate)}
                />
              )}
              {todayData.heartRateVariability && (
                <HeartRateAVGCard
                  title={'Heart rate variability average'}
                  dateTime={'Today'}
                  value={String(todayData.heartRateVariability)}
                />
              )}
            </>
          )}
          <CText size={'lg_light'}>Past 7 days</CText>
          {heartRate
            .filter((data) => data.date !== today)
            .map((data) => (
              <View key={data.date}>
                {data.heartRate && (
                  <HeartRateAVGCard
                    title={'Heart rate average'}
                    dateTime={`${data.day} ${data.date}`}
                    value={String(data.heartRate)}
                  />
                )}
                {data.heartRateVariability && (
                  <HeartRateAVGCard
                    title={'Heart rate variability average'}
                    dateTime={`${data.day} ${data.date}`}
                    value={String(data.heartRateVariability)}
                  />
                )}
              </View>
            ))}
        </View>
      )}
    </Screen>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    gap: 12,
    paddingTop: 20,
    paddingHorizontal: 20,
    paddingBottom: 50,
  },
  wrapper: {
    justifyContent: 'center',
    width: '100%',
    gap: 20,
  },
});
