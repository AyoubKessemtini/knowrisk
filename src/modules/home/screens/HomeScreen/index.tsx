import { CButton } from '@components/Buttons/CButton';
import { CText } from '@components/CText';
import { ControlledInput } from '@components/ControlledInput';
import { Screen } from '@components/Screen';
import { Toggle } from '@components/Toggle';
import { zodResolver } from '@hookform/resolvers/zod';
import { useCTheme } from '@hooks/useCTheme';
import { FooStackRoutes, RootStackRoutes } from '@navigators/routes';
import { TabStackScreenProps } from '@navigators/stacks/TabNavigator';
import { useAppDispatch } from '@store/index';
import { LocaleActions } from '@store/localeSlice';
import React from 'react';
import { useForm } from 'react-hook-form';
import { StyleSheet } from 'react-native';
import { SheetManager } from 'react-native-actions-sheet';
import { Languages } from 'src/i18n/types';
import { SheetsTypes } from 'src/lib/sheets';
import { HomeScheme, homeScheme } from './home.scheme';

export const Home = ({
  navigation,
}: TabStackScreenProps<'home'>): JSX.Element => {
  const { isLightTheme, toggleTheme } = useCTheme();
  const dispatch = useAppDispatch();

  const onPressHandler = () => {
    navigation.navigate(RootStackRoutes.FOO_STACK, {
      screen: FooStackRoutes.FOO_SCREEN,
    });
  };

  const { control, handleSubmit } = useForm<HomeScheme>({
    defaultValues: { username: undefined },
    resolver: zodResolver(homeScheme),
  });

  const onSubmit = (formData: HomeScheme) => {
    console.log(formData);
  };

  const handleChangeLocale = (locale: Languages) => {
    dispatch(LocaleActions.setLocale(locale));
  };

  const actionSheetHandler = async () => {
    const callback = await SheetManager.show(SheetsTypes.GOOD_SHEET, {
      payload: { text: 'Wow, text payload' },
    });

    if (callback?.text) {
      return console.log(callback.text);
    }

    if (callback?.shouldNavigate) {
      return navigation.navigate(RootStackRoutes.FOO_STACK, {
        screen: FooStackRoutes.FOO_SCREEN,
      });
    }
  };

  return (
    <Screen withoutBottomEdge containerStyles={styles.container}>
      <CText size="xl_bold" isCentered text="home.foo" />

      <ControlledInput control={control} name="username" />
      <CButton
        buttonType="success"
        text="common.apply"
        onPress={handleSubmit(onSubmit)}
      />
      <CButton
        mt={16}
        buttonType="primary"
        text="common.continue"
        onPress={onPressHandler}
      />
      <CButton
        buttonType="danger"
        text="common.showsheet"
        onPress={actionSheetHandler}
      />

      <CButton
        mt={16}
        buttonType="secondary"
        text="language.english"
        onPress={() => handleChangeLocale(Languages.EN)}
      />
      <CButton
        buttonType="secondary"
        text="language.french"
        onPress={() => handleChangeLocale(Languages.FR)}
      />

      <Toggle activated={isLightTheme} onPressHandler={toggleTheme} />
    </Screen>
  );
};

const styles = StyleSheet.create({
  container: {
    gap: 20,
  },
});
