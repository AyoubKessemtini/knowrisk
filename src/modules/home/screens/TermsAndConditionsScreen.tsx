import React, { useState } from 'react';
import {
  View,
  ActivityIndicator,
  TouchableOpacity,
  Text,
  StyleSheet,
  SafeAreaView,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { WebView } from 'react-native-webview';

export const TermsAndConditionsScreen = (): JSX.Element => {
  const [loading, setLoading] = useState(true);
  const navigation = useNavigation();

  const handleClose = () => {
    // Use navigation.dismiss() if the screen is presented modally
    // navigation.dismiss();

    // If navigation.dismiss() doesn't work, use navigation.goBack()
    navigation.goBack(); // Retourne à l’écran précédent
  };

  return (
    <SafeAreaView style={{ flex: 1 }}>

    <View style={{ flex: 1 }}>
      {/* En-tête personnalisé */}
      
      <View style={styles.header}>
        <TouchableOpacity onPress={handleClose} style={styles.closeButton}>
          <Text style={styles.closeButtonText}>Close</Text>
        </TouchableOpacity>
        <Text style={styles.headerTitle}>Terms & Conditions</Text>
      </View>

      {/* Indicateur de chargement */}
      {loading && (
        <ActivityIndicator
          size="large"
          color="#0000ff"
          style={styles.activityIndicator}
        />
      )}

      {/* WebView */}
      <WebView
        source={{ uri: 'https://www.knowlepsy.com/personal-data-policy' }}
        style={{ flex: 1 }}
        onLoadEnd={() => setLoading(false)}
      />
    </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  header: {
    height: 60,
    backgroundColor: '#f5f5f5',
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 10,
    elevation: 4, // Pour une ombre sur Android
    shadowColor: '#000', // Pour une ombre sur iOS
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
  },
  closeButton: {
    padding: 10,
  },
  closeButtonText: {
    color: '#007AFF',
    fontSize: 16,
    fontWeight: 'bold', // Make the text bold
  },
  headerTitle: {
    flex: 1,
    textAlign: 'center',
    fontSize: 18,
    fontWeight: '600',
  },
  activityIndicator: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    marginLeft: -25, // Pour centrer l’indicateur
    marginTop: -25,
  },
});
