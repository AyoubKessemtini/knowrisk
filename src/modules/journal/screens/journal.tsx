import React, { useState } from 'react';
import { Alert, StyleSheet, View } from 'react-native';
import { Screen } from '@components/Screen';
import { QuestionComponent } from '@components/questioncard';
import { CText } from '@components/CText';
import { Colors } from '@constants/Colors';
import { Header } from '@components/Headers/Header.tsx';
import { CButton } from '@components/Buttons/CButton.tsx';
import { DateSelector } from '@components/DatePicker/DatePicker.tsx';
import { formatStringDate } from '@hooks/useDateFormatter.ts';

export const JournalScreen: React.FC = () => {
  const [selectedDate, setSelectedDate] = useState(new Date());
  const questionsData = [
    {
      category: 'Stress',
      questions: [
        'Did you experience stress today?',
        'Did you have any emotional distress or anxiety today?',
        'Did you notice any changes in your mood or mental state today?',
      ],
    },
    {
      category: 'Nutrition',
      questions: [
        'Did you consume any caffeine today?',
        'Did you consume any alcohol today?',
        'Did you consume any new foods or drinks today?',
      ],
    },
    {
      category: 'Sleep',
      questions: [
        'Did you experience any unusual lack of sleep last night?',
        'Were you feeling particularly tired or fatigued today?',
      ],
    },
    {
      category: 'Physical Activity',
      questions: ['Did you engage in any intense physical activity today?'],
    },
    {
      category: 'Environment',
      questions: [
        'Were you exposed to flashing or bright lights today?',
        'Were you in a crowded or noisy environment today?',
      ],
    },
    {
      category: 'Medication',
      questions: [
        'Did you miss any doses of your prescribed medication today?',
      ],
    },
    {
      category: 'Headaches/Migraines',
      questions: ['Did you experience a headache or migraine today?'],
    },
    {
      category: 'Screen Time',
      questions: [
        'Did you spend more time than usual on screens (phone, computer, etc.) today?',
      ],
    },
  ];
  const handleDateChange = (newDate: Date) => {
    if (newDate > new Date()) {
      Alert.alert(
        'Date error',
        `Questions of (${formatStringDate(newDate)}) are not available.`,
      );
    } else {
      setSelectedDate(newDate);
    }
  };
  const allQuestions = questionsData.flatMap((section) =>
    section.questions.map((question) => ({
      question,
      category: section.category,
    })),
  );
  const shuffleArray = (array: { question: string; category: string }[]) => {
    return array.sort(() => Math.random() - 0.5);
  };
  const randomQuestions = shuffleArray(allQuestions).slice(0, 4);
  return (
    <Screen
      fullscreen
      withoutTopEdge
      noHorizontalPadding
      containerStyles={styles.container}
    >
      <Header
        hasBackButton
        useCustomBackButton
        text="common.journal"
        textColor="black"
      />
      <View style={styles.wrapper}>
        <DateSelector
          initialDate={selectedDate}
          onDateChange={handleDateChange}
        />
        <CText
          style={{ textAlign: 'center' }}
          mt={20}
          mb={20}
          size={'md_light'}
          color={'grey3'}
        >
          {' '}
          What happened {formatStringDate(selectedDate)}
        </CText>
        {randomQuestions.map((item, idx) => (
          <View key={idx} style={styles.sectionContainer}>
            <CText size="lg_bold" color="purple">
              {item.category}
            </CText>
            <QuestionComponent key={idx} question={item.question} />
          </View>
        ))}
      </View>
      <CButton
        text="buttons.save"
        onPress={() => {
          console.log('save journal');
        }}
      />
    </Screen>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    paddingTop: 20,
    paddingHorizontal: 20,
    paddingBottom: 50,
  },
  wrapper: {
    justifyContent: 'center',
    width: '100%',
  },
  sectionContainer: {
    marginTop: 7,
    marginBottom: 7,
    borderWidth: 1,
    borderColor: Colors.lightPurple,
    borderRadius: 10,
    padding: 10,
  },
});
