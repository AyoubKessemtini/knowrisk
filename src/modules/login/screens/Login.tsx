import React, { useEffect } from 'react';
import {
  Alert,
  ImageBackground,
  KeyboardAvoidingView,
  Platform,
  Pressable,
  ScrollView,
  StyleSheet,
  View,
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { useForm } from 'react-hook-form';
import { zodResolver } from '@hookform/resolvers/zod';
import { useNavigation } from '@react-navigation/native';
import { CButton } from '@components/Buttons/CButton';
import { ControlledInput } from '@components/ControlledInput';
import { PassTextInput } from '@components/Inputs/Pass-Text-Input';
import { CText } from '@components/CText';
import { LoginScheme, loginScheme } from '../../../schemes/login.scheme';
import ImageAssets from '@assets/images';
import { Colors } from '@constants/Colors';
import { AuthActions } from '@store/authSlice';
import { OnboardingStackRoutes, RootStackRoutes } from '@navigators/routes';
import { RootState } from '@store/index.ts';
import { OnboardingStackScreenProps } from '@navigators/stacks/OnboardingNavigator.tsx';

export const LoginScreen = ({}: OnboardingStackScreenProps<'LoginScreen'>) => {
  const navigation = useNavigation();
  const dispatch = useDispatch();

  const { control, handleSubmit } = useForm<LoginScheme>({
    defaultValues: { email: '', password: '' },
    resolver: zodResolver(loginScheme),
  });

  const loading = useSelector((state: RootState) => state.auth.loading);
  const error = useSelector((state: RootState) => state.auth.errorLogin);

  useEffect(() => {
    dispatch(AuthActions.resetErrorLogin());
    return () => {
      dispatch(AuthActions.resetErrorLogin());
    };
  }, [dispatch]);

  useEffect(() => {
    if (error) {
      Alert.alert('Login Failed', error, [
        {
          text: 'OK',
          onPress: () => dispatch(AuthActions.resetErrorLogin()),
        },
      ]);
    }
  }, [error, dispatch]);

  const onPressHandler = (formData: LoginScheme) => {
    dispatch(AuthActions.resetErrorLogin());
    dispatch(AuthActions.loginRequest(formData));
  };

  return (
    <ImageBackground source={ImageAssets.LOGIN_BG} style={styles.container}>
      <KeyboardAvoidingView
        style={styles.flex}
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        keyboardVerticalOffset={0}
      >
        <ScrollView
          contentContainerStyle={styles.scrollContent}
          showsVerticalScrollIndicator={false}
        >
          <View style={styles.flexed} />
          <CText
            mt={38}
            text="onboarding.login_description"
            color="white"
            size="xl_bold"
          />
          <CText
            mt={5}
            text="onboarding.login_to"
            color="white"
            size="md_medium"
          />
          <CText
            mt={15}
            text="onboarding.email_address"
            color="white"
            size="md"
          />
          <ControlledInput
            placeholderText="common.mail"
            placeholderTextColor={Colors.grey2}
            control={control}
            name="email"
            borderColor="white"
            backgroundColor={Colors.lightPink}
            textStyle={{ color: Colors.deepPurple }}
          />
          <CText mt={15} text="onboarding.password" color="white" size="md" />
          <PassTextInput name="password" control={control} />
          <CButton
            mt={15}
            buttonType="primary"
            text="onboarding.login"
            onPress={handleSubmit(onPressHandler)}
          />
          <Pressable
            onPress={() =>
              navigation.navigate(RootStackRoutes.ONBOARDING_STACK, {
                screen: OnboardingStackRoutes.FORGOT_PASSWORD_SCREEN,
              })
            }
          >
            <CText
              isCentered
              text="onboarding.forgotPass"
              color="deepPurple"
              size="md"
              mt={10}
            />
          </Pressable>
          <CText isCentered size="md">
            <CText
              text="onboarding.not_member"
              color="white"
              size="md"
              mb={30}
            />
            <Pressable
              onPress={() =>
                navigation.navigate(RootStackRoutes.ONBOARDING_STACK, {
                  screen: OnboardingStackRoutes.SIGNUP_SCREEN,
                })
              }
            >
              <CText text="onboarding.register" color="deepPurple" size="md" />
            </Pressable>
          </CText>
        </ScrollView>
      </KeyboardAvoidingView>
    </ImageBackground>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  flex: {
    flex: 1,
  },
  scrollContent: {
    paddingTop: 300,
    paddingHorizontal: 20,
    gap: 15,
  },
  flexed: {
    gap: 12,
  },
});
