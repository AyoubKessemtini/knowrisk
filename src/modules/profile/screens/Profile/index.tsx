import { Screen } from '@components/Screen';
import { Colors } from '@constants/Colors';
import { TabStackScreenProps } from '@navigators/stacks/TabNavigator';
import React from 'react';
import { StyleSheet, Text } from 'react-native';

export const Profile = ({}: TabStackScreenProps<'profile'>): JSX.Element => {
  return (
    <Screen containerStyles={styles.container}>
      <Text style={styles.textStyle}>PROFILE</Text>
    </Screen>
  );
};

const styles = StyleSheet.create({
  textStyle: {
    color: Colors.black,
    fontWeight: 'bold',
    fontSize: 30,
  },
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
});
