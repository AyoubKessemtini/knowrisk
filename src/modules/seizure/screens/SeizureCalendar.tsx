import React, { useEffect, useState } from 'react';
import { StyleSheet, View, ScrollView } from 'react-native';
import { Screen } from '@components/Screen';
import { EventCalendar } from '@components/EventCalendar';
import { RootStackScreenProps } from '@navigators/stacks/RootNavigator';
import { CText } from '@components/CText';
import { Header } from '@components/Headers/Header';
import { PersistenceStorage } from '@storage/index';
import { KEYS } from '@storage/Keys';
import { RootState } from '@store/index';
import { seizureActions } from '@store/sagas/seizureSlice';
import { useDispatch, useSelector } from 'react-redux';

type EventData = {
  id: string;
  title: string;
  date: string;
  description: string;
  timeFrom: string;
  timeTo: string;
};

export const SeizureForecastScreen: React.FC<
  RootStackScreenProps<'SeizureForecastScreen'>
> = () => {
  const dispatch = useDispatch();
  const { events, loading, error } = useSelector(
    (state: RootState) => state.seizureForecast,
  );
  const [userId, setUserId] = useState<string | null>(null);
  const [selectedDate, setSelectedDate] = useState<string | null>(null);

  // Marked dates configuration
  const markedDates = events.reduce(
    (acc, event) => {
      acc[event.date] = {
        selected: true,
        marked: true,

        selectedColor: event.type === 'Reported' ? '#e74c3c' : '#2ecc71',
      };
      return acc;
    },
    {} as Record<
      string,
      { selected: boolean; marked: boolean; selectedColor: string }
    >,
  );

  // Populate `eventsData` directly from `events`
  const eventsData: EventData[] = events.map((event) => ({
    id: '',
    title: '',
    date: event.date,
    description: '',
    timeFrom: event.time_from || 'N/A', // Default to 'N/A' if not provided
    timeTo: event.time_to || 'N/A', // Default to 'N/A' if not provided
  }));

  useEffect(() => {
    const fetchUserId = async () => {
      const userData = await PersistenceStorage.getItem(KEYS.USER_DATA);
      if (userData) {
        const user = JSON.parse(userData);
        setUserId(user.id);
      }
    };
    fetchUserId();
  }, []);

  useEffect(() => {
    if (userId) {
      const currentMonth = new Date().toISOString().slice(0, 7);
      dispatch(
        seizureActions.fetchSeizureForecastRequest({
          id: userId,
          month: currentMonth,
        }),
      );
    }
  }, [dispatch, userId]);

  // Determine which events to display based on selectedDate
  const filteredEvents = selectedDate
    ? eventsData.filter((event) => event.date === selectedDate)
    : eventsData; // Show all events initially

  return (
    <Screen
      fullscreen
      withoutTopEdge
      noHorizontalPadding
      containerStyles={styles.container}
    >
      <Header hasBackButton text="common.seizure" />

      {/* Pass the onDayPress handler to EventCalendar */}
      <EventCalendar
        markedDates={markedDates}
        onDayPress={(day) => {
          console.log('Selected date:', day.dateString);
          setSelectedDate(day.dateString); // Update the selected date
        }}
      />

      <CText style={styles.sectionTitle}>Seizure Events</CText>
      <ScrollView style={styles.eventsContainer}>
        {filteredEvents.length > 0 ? (
          filteredEvents.map((item, index) => (
            <View
              key={`${item.date}-${item.id || index}`}
              style={styles.eventItem}
            >
              <View style={styles.eventIcon}>
                <CText style={styles.iconText}>âš¡</CText>
              </View>
              <View style={styles.eventDetails}>
                <CText style={styles.eventDate}>{item.date}</CText>
                <CText style={styles.eventTime}>
                  {`From: ${item.timeFrom || 'N/A'} To: ${item.timeTo || 'N/A'}`}
                </CText>
                <CText style={styles.eventDescription}>
                  {item.description || ''}
                </CText>
              </View>
            </View>
          ))
        ) : (
          <CText style={styles.emptyText}>
            No events available for this date
          </CText>
        )}
      </ScrollView>
    </Screen>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    gap: 12,
    paddingTop: 20,
    paddingHorizontal: 20,
  },
  sectionTitle: {
    fontSize: 20,
    fontWeight: '700',
    marginTop: 16,
    marginBottom: 12,
    color: '#34495e',
  },
  eventsContainer: {
    width: '100%',
  },
  eventItem: {
    flexDirection: 'row',
    backgroundColor: '#fce4ec',
    padding: 14,
    borderRadius: 10,
    marginBottom: 12,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.2,
    shadowRadius: 1.5,
  },
  eventIcon: {
    marginRight: 12,
    width: 40,
    height: 40,
    borderRadius: 20,
    backgroundColor: '#e74c3c',
    justifyContent: 'center',
    alignItems: 'center',
  },
  iconText: {
    color: '#fff',
    fontSize: 18,
  },
  eventDetails: {
    flex: 1,
  },
  eventDate: {
    fontSize: 14,
    color: '#7f8c8d',
    marginBottom: 2,
  },
  eventTime: {
    fontSize: 13,
    color: '#2c3e50',
    marginBottom: 2,
  },
  eventDescription: {
    fontSize: 13,
    color: '#7f8c8d',
  },
  emptyText: {
    textAlign: 'center',
    color: '#7f8c8d',
    marginTop: 20,
  },
});
