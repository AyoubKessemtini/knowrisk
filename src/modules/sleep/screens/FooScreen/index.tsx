import { CText } from '@components/CText';
import { Colors } from '@constants/Colors';
import { FooStackScreenProps } from '@navigators/stacks/FooNavigator';
import React from 'react';
import { Button, StyleSheet, View } from 'react-native';

interface Props extends FooStackScreenProps<'FooScreen'> {
  someProp?: boolean;
}

export const FooScreen = ({ navigation }: Props): JSX.Element => {
  return (
    <View style={styles.container}>
      <CText text="common.continue" />
      <Button onPress={() => navigation.navigate('BooScreen')} title="CLICK" />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.white,
  },
});
