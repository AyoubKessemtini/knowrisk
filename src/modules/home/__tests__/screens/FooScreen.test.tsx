import { FooScreen } from '@modules/home/screens/FooScreen';
import { store } from '@store/index';
import { render } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux';

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
  },
  route: {},
});

describe('[MODULE: HOME] | FooScreen', () => {
  let props: any;

  beforeEach(() => {
    props = createTestProps();
  });

  it('renders correctly', () => {
    const foo = render(
      <Provider store={store}>
        <FooScreen {...props} />
      </Provider>,
    );
    expect(foo).toMatchSnapshot();
  });
});
