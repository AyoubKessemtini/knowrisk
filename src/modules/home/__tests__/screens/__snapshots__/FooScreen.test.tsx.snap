// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[MODULE: HOME] | FooScreen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "GoogleSans-Regular",
        },
        {
          "fontSize": 14,
        },
        {
          "fontFamily": "GoogleSans-Medium",
        },
        {
          "color": undefined,
        },
        false,
        false,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    French Continue
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        CLICK
      </Text>
    </View>
  </View>
</View>
`;
